src/tests/patient-controller-test.test.ts

```diff
@@ -0,0 +1,81 @@
+import { Request, Response, NextFunction } from 'express';
+import * as patientController from '../controllers/patient-controller';
+import PatientModel from '../models/patient';
+
+// Mock the PatientModel
+jest.mock('../models/patient');
+
+describe('Patient Controller - Unit Tests', () => {
+  const mockRequest = {} as Request;
+  let mockResponse: Response;
+  let mockNext: NextFunction;
+
+  beforeEach(() => {
+    mockNext = jest.fn();
+    mockResponse = {
+      json: jest.fn(),
+      status: jest.fn().mockReturnThis(), // Properly mock status chaining
```

Comment:
I can see what's happening just by looking at the code. We should only add comments that are useful. For example to explain some complex logic